src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -1005,6 +1030,92 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
             JasmineGraphServer *jasmineServer = new JasmineGraphServer();
             bool isSpawned = jasmineServer->spawnNewWorker(host,port,dataPort,profile,masterHost,enableNmon);
 
+        } else if (line.compare(SLA) == 0) {
+            int result_wr = write(connFd, COMMAND.c_str(), COMMAND.size());
+            if(result_wr < 0) {
+                frontend_logger.log("Error writing to socket", "error");
+            }
+            result_wr = write(connFd, "\r\n", 2);
+            if(result_wr < 0) {
+                frontend_logger.log("Error writing to socket", "error");
+            }
+
+            char category[FRONTEND_DATA_LENGTH];
+            bzero(category, FRONTEND_DATA_LENGTH + 1);
+
+            read(connFd, category, FRONTEND_DATA_LENGTH);
+
+            string command_info(category);
+
+            Utils utils;
+            command_info = utils.trim_copy(command_info, " \f\n\r\t\v");
+            frontend_logger.log("Data received: " + command_info, "info");
+
+            std::vector<vector<pair<string, string>>> categoryResults = perfSqlite.runSelect(
+                    "SELECT id FROM sla_category where command='" + command_info + "';");
+
+            string slaCategoryIds;
+
+            for (std::vector<vector<pair<string, string>>>::iterator i = categoryResults.begin(); i != categoryResults.end(); ++i) {
+                for (std::vector<pair<string, string>>::iterator j = (i->begin()); j != i->end(); ++j) {
+                    slaCategoryIds = slaCategoryIds + "'" + j->second + "',";
+                }
+            }
+
+            string adjustedIdList = slaCategoryIds.substr(0, slaCategoryIds.size()-1);
```

Comment:
Lets have space between `slaCategoryIds.size()-1`. I.e., lets change `slaCategoryIds.size()-1` to `slaCategoryIds.size() - 1`